setup.py

```diff
@@ -25,9 +25,15 @@
         "./lib/ultrajsonenc.c",
         "./lib/ultrajsondec.c",
     ],
-    include_dirs=["./python", "./lib", "./deps/double-conversion/double-conversion"],
+    include_dirs=["./python", "./lib"]
+    + environ.get(
+        "UJSON_BUILD_DC_INCLUDES",
+        "./deps/double-conversion/double-conversion",
+    ).split(),
     extra_compile_args=["-D_GNU_SOURCE"],
-    extra_link_args=["-lstdc++", "-lm"] + strip_flags,
+    extra_link_args=["-lstdc++", "-lm"]
+    + environ.get("UJSON_BUILD_DC_LIBS", "").split()
```

Comment:
Normally this kind of thing is split by `os.pathsep`. I'd be very surprised if a path with spaces ever did need to be added there. With `shlex.split()` would at least let you quote or backslash escape a space if need be.

Reply:
Thanks for the suggestions.

I tend to think that the `shlex.split()` approach is appropriate for `UJSON_BUILD_DC_LIBS`, since `LDFLAGS`-style environment variables are usually like `-L/usr/lib/foo -lbar -lbat`. There’s some risk of surprising magic by doing this, but it pretty closely resembles the behavior of traditional build systems that end up doing something like `${CC} ${CFLAGS} ${LDFLAGS} …` with unquoted shell interpolation.

```python
shlex.split(environ.get("UJSON_BUILD_DC_LIBS", ""))
```

Meanwhile, the proposed `UJSON_BUILD_DC_INCLUDES` is a list-of-paths like `PATH` rather than a set of `CFLAGS`-style compiler flags, so splitting on `os.pathsep`, i.e. `:`, does seem reasonable there.

```python
environ.get("UJSON_BUILD_DC_INCLUDES", "./deps/double-conversion/double-conversion").split(pathsep)
```

Sound good?